game-core/src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -286,10 +285,10 @@ private static String validateCanal(
     // Check each unit 1 by 1 to see if they can move through necessary canals on route
     String result = null;
     final Set<Unit> unitsThatFailCanal = new HashSet<>();
+    final Set<Unit> setWithNull = new HashSet<>();
+    setWithNull.add(null);
```

Comment:
Not to sound mean or something, but this seems like a prime example to my why the "old" methods still have their use.
So unless you find a way to avoid null usage in every place like this (why do we even need a singleton, could an empty set work as well?) I don't think this is a step forward

Reply:
Also hashset is mutable, whereas the singleton is not.

Reply:
The mutability of collections depends on the contents. Singleton is unmodifiable, but it's not necessarily immutable. Favoring immutability is much more about removing setters.

Reply:
To this example, Unit is mutable, the singleton collection wrapper around it does not change that.